packages/@aws-cdk/aws-stepfunctions-tasks/README.md

```diff
@@ -9,4 +9,668 @@
+    ],
+    actions: [
+      'sts:AssumeRole',
+    ],
+  });
+)
+
+new sfn.Task(stack, 'Create Cluster', {
+  task: new tasks.EmrCreateCluster({
+    instances: {},
+    clusterRole,
+    name: sfn.TaskInput.fromDataAt('$.ClusterName').value,
+    serviceRole,
+    autoScalingRole,
+    integrationPattern: sfn.ServiceIntegrationPattern.FIRE_AND_FORGET,
+  }),
+});
+```
+
+##### Termination Protection
+
+Locks a cluster (job flow) so the EC2 instances in the cluster cannot be
+terminated by user intervention, an API call, or a job-flow error.
+
+Corresponds to the [`setTerminationProtection`](https://docs.aws.amazon.com/step-functions/latest/dg/connect-emr.html) API in EMR.
+
+```ts
+new sfn.Task(stack, 'Task', {
+  task: new tasks.EmrSetClusterTerminationProtection({
+    clusterId: 'ClusterId',
+    terminationProtected: false,
+  }),
+});
+```
+
+##### Terminate Cluster
+
+Shuts down a cluster (job flow).
+Corresponds to the [`terminateJobFlows`](https://docs.aws.amazon.com/emr/latest/APIReference/API_TerminateJobFlows.html) API in EMR.
+
+```ts
+new sfn.Task(stack, 'Task', {
+  task: new tasks.EmrTerminateCluster({
+    clusterId: 'ClusterId'
+  }),
+});
+```
+
+##### Add Step
+
+Adds a new step to a running cluster.
+Corresponds to the [`addJobFlowSteps`](https://docs.aws.amazon.com/emr/latest/APIReference/API_AddJobFlowSteps.html) API in EMR.
+
+```ts
+new sfn.Task(stack, 'Task', {
+  task: new tasks.EmrAddStep({
+    clusterId: 'ClusterId',
+    name: 'StepName',
+    jar: 'Jar',
+    actionOnFailure: tasks.ActionOnFailure.CONTINUE,
+  }),
+});
+```
+
+##### Cancel Step
+
+Cancels a pending step in a running cluster.
+Corresponds to the [`cancelSteps`](https://docs.aws.amazon.com/emr/latest/APIReference/API_CancelSteps.html) API in EMR.
+
+```ts
+new sfn.Task(stack, 'Task', {
+  task: new tasks.EmrCancelStep({
+    clusterId: 'ClusterId',
+    stepId: 'StepId',
+  }),
+});
+```
+
+##### Modify Instance Fleet
+
+Modifies the target On-Demand and target Spot capacities for the instance
+fleet with the specified InstanceFleetName.
+
+Corresponds to the [`modifyInstanceFleet`](https://docs.aws.amazon.com/emr/latest/APIReference/API_ModifyInstanceFleet.html) API in EMR.
+
+```ts
+new sfn.Task(stack, 'Task', {
+  task: new tasks.EmrModifyInstanceFleetByName({
+    clusterId: 'ClusterId',
+    instanceFleetName: 'InstanceFleetName',
+    targetOnDemandCapacity: 2,
+    targetSpotCapacity: 0,
+  }),
+});
+```
+
+##### Modify Instance Group
+
+Modifies the number of nodes and configuration settings of an instance group.
+
+Corresponds to the [`modifyInstanceGroups`](https://docs.aws.amazon.com/emr/latest/APIReference/API_ModifyInstanceGroups.html) API in EMR.
+
+```ts
+new sfn.Task(stack, 'Task', {
+  task: new tasks.EmrModifyInstanceGroupByName({
+    clusterId: 'ClusterId',
+    instanceGroupName: sfn.Data.stringAt('$.InstanceGroupName'),
+    instanceGroup: {
+      instanceCount: 1,
+    },
+  }),
+});
+```
+
+#### Evaluate Expression
```

Comment:
This is a useful one, and not specific to any service integration. 
Wonder if we should bump this to the top of the list.